File: java/src/test/java/com/agilesparks/rubikscube/utils/FaceTest.java
package com.agilesparks.rubikscube.utils;


import static org.junit.Assert.*;

import org.junit.Test;

import com.agilesparks.rubikscube.utils.Face;

public class FaceTest {
    @Test
    public void testFaceGetIntGetChar() {
        Face myFace = Face.BOTTOM;
        assertEquals(1, myFace.getInt());
        assertEquals('D', myFace.getIntOfChar());
        myFace = Face.FRONT;
        assertEquals(4, myFace.getInt());
        assertEquals('F', myFace.getIntOfChar());
        myFace = Face.RIGHT;
        assertEquals(2, myFace.getInt());
        assertEquals('R', myFace.getIntOfChar());
        myFace = Face.TOP;
        assertEquals(0, myFace.getInt());
        assertEquals('U', myFace.getIntOfChar());
        myFace = Face.BACK;
        assertEquals(5, myFace.getInt());
        assertEquals('B', myFace.getIntOfChar());
        myFace = Face.LEFT;
        assertEquals(3, myFace.getInt());
        assertEquals('L', myFace.getIntOfChar());
    };

    @Test
    public void testFaceEquals() {
        Face myFace = Face.BOTTOM;
        assertEquals(Face.BOTTOM, myFace);
        assertEquals(true, myFace==Face.BOTTOM);
    }

    @Test
    public void testFaceGetOpposite() {
        Face myFace = Face.BOTTOM;
        assertEquals(Face.TOP, myFace.getOpposite());
        myFace = Face.FRONT;
        assertEquals(Face.BACK, myFace.getOpposite());
        myFace = Face.RIGHT;
        assertEquals(Face.LEFT, myFace.getOpposite());
        myFace = Face.TOP;
        assertEquals(Face.BOTTOM, myFace.getOpposite());
        myFace = Face.BACK;
        assertEquals(Face.FRONT, myFace.getOpposite());
        myFace = Face.LEFT;
        assertEquals(Face.RIGHT, myFace.getOpposite());
    }

}

File: python/Tests/utils/FaceTest.py

using utils




    [TestClass]
    class FaceTest
    
        [TestMethod]
        def testFaceGetIntGetChar():
            Face myFace = Face.BOTTOM
            Assert.AreEqual(1,(int) myFace)
            Assert.AreEqual('D', FaceHandler.getCharValue(myFace))
            myFace = Face.FRONT
            Assert.AreEqual(4, (int)myFace)
            Assert.AreEqual('F', FaceHandler.getCharValue(myFace))
            myFace = Face.RIGHT
            Assert.AreEqual(2, (int)myFace)
            Assert.AreEqual('R', FaceHandler.getCharValue(myFace))
            myFace = Face.TOP
            Assert.AreEqual(0, (int)myFace)
            Assert.AreEqual('U', FaceHandler.getCharValue(myFace))
            myFace = Face.BACK
            Assert.AreEqual(5, (int)myFace)
            Assert.AreEqual('B', FaceHandler.getCharValue(myFace))
            myFace = Face.LEFT
            Assert.AreEqual(3, (int)myFace)
            Assert.AreEqual('L', FaceHandler.getCharValue(myFace))
        

        [TestMethod]
        def testFaceEquals():
            Face myFace = Face.BOTTOM
            Assert.AreEqual(Face.BOTTOM, myFace)
            Assert.AreEqual(true, myFace == Face.BOTTOM)
        

        [TestMethod]
        def testFaceGetOpposite():
            Face myFace = Face.BOTTOM
            Assert.AreEqual(Face.TOP, FaceHandler.getOpposite(myFace))
            myFace = Face.FRONT
            Assert.AreEqual(Face.BACK, FaceHandler.getOpposite(myFace))
            myFace = Face.RIGHT
            Assert.AreEqual(Face.LEFT, FaceHandler.getOpposite(myFace))
            myFace = Face.TOP
            Assert.AreEqual(Face.BOTTOM, FaceHandler.getOpposite(myFace))
            myFace = Face.BACK
            Assert.AreEqual(Face.FRONT, FaceHandler.getOpposite(myFace))
            myFace = Face.LEFT
            Assert.AreEqual(Face.RIGHT, FaceHandler.getOpposite(myFace))
        

    
